front/pages/w/[wId]/ds/index.tsx

```diff
@@ -97,30 +97,20 @@ export const getServerSideProps: GetServerSideProps<{
   let allDataSources = await getDataSources(auth);
   const dataSources = allDataSources.filter((ds) => !ds.connector);
   const managedDataSources = allDataSources.filter((ds) => ds.connector);
-
-  let connectorStatuses = managedDataSources.map(() => "failed");
-  try {
-    connectorStatuses = (
-      await Promise.all(
-        managedDataSources.map(({ connector }) =>
-          ConnectorsAPI.getSyncStatus(connector!.id)
-        )
-      )
-    ).map((s): ConnectorSyncStatus => (s.isErr() ? "failed" : "succeeded"));
-  } catch (e) {
-    // Probably means `connectors` is down, we log but don't fail to avoid a 500 when just
-    // displaying the datasources (eventual actions will fail but a 500 just at display is not
-    // desirable). When that happens the managed data sources are shown as failed.
-    logger.error({ error: e }, "Failed to retrieve connector status");
-  }
-
-  const connectorStatusByProvider: Record<
-    ConnectorProvider,
-    ConnectorSyncStatus
-  > = managedDataSources.reduce(
-    (acc, ds, i) =>
-      Object.assign(acc, { [ds.connector!.provider]: connectorStatuses[i] }),
-    {} as Record<ConnectorProvider, ConnectorSyncStatus>
+  const provider2Connector = await Promise.all(
+    managedDataSources.map(async (mds) => {
+      const statusRes = await ConnectorsAPI.getConnector(mds.connector!.id);
```

Comment:
Maybe I'm missing something but since you removed the try/catch, it means that if `connectors` is down we 500 here the entire page no? If that's the case (easy to test) please reintroduce the try/catch + comment?

Reply:
Ok my bad. This part of the code was very confusing to me because we were managing:
- "ManagedDataSource" as an array of the providers we support (aka connectors).
- "allDataSources" as an array of user created data sources.
- "data source to status via the provider"
- The sub object `connector` in `DataSourceType`, which only has two properties (`id` and `provider`).

It was not clear to me which object was what, so I ended up removing that code by mistake.
I will rename the object "ManagedDataSource" to `CONNECTORS_LIST`, and put back the try/catch.